src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -4205,6 +4205,11 @@ public void visitRecordPattern(JCRecordPattern tree) {
         List<Type> expectedRecordTypes;
         if (site.tsym.kind == Kind.TYP && ((ClassSymbol) site.tsym).isRecord()) {
             ClassSymbol record = (ClassSymbol) site.tsym;
+            if (record.type.getTypeArguments().nonEmpty()) {
```

Comment:
There is a `Type::isRaw()` - I supposed you tried that one? Doesn't it do what you want?